src/ray/gcs/gcs_server/test/gcs_server_rpc_test.cc

```diff
@@ -127,6 +127,47 @@ class GcsServerTest : public ::testing::Test {
     AsyncCall(call_function, timeout_ms_);
   }
 
+  void TestRegisterNodeInfo(const rpc::RegisterNodeInfoRequest &request) {
+    auto call_function = [this, request](std::promise<bool> &promise) {
+      client_->RegisterNodeInfo(
+          request,
+          [&promise](const Status &status, const rpc::RegisterNodeInfoReply &reply) {
+            RAY_CHECK_OK(status);
+            promise.set_value(true);
+          });
+    };
+    AsyncCall(call_function, timeout_ms_);
+  }
+
+  void TestUnregisterNodeInfo(const rpc::UnregisterNodeInfoRequest &request) {
+    auto call_function = [this, request](std::promise<bool> &promise) {
+      client_->UnregisterNodeInfo(
+          request,
+          [&promise](const Status &status, const rpc::UnregisterNodeInfoReply &reply) {
+            RAY_CHECK_OK(status);
+            promise.set_value(true);
+          });
+    };
+    AsyncCall(call_function, timeout_ms_);
+  }
+
+  void GetAllNodesInfo(std::vector<rpc::GcsNodeInfo> &node_infos) {
+    node_infos.clear();
+    rpc::GetAllNodesInfoRequest request;
+    auto call_function = [this, request, &node_infos](std::promise<bool> &promise) {
+      client_->GetAllNodesInfo(
+          request, [&promise, &node_infos](const Status &status,
+                                           const rpc::GetAllNodesInfoReply &reply) {
+            RAY_CHECK_OK(status);
+            promise.set_value(true);
```

Comment:
The promise should be set after node_infos be filled.

Reply:
fixed